common/web/gesture-recognizer/src/gestureRecognizerConfiguration.ts

```diff
@@ -18,6 +25,45 @@ namespace com.keyman.osk {
      */
     readonly targetRoot: HTMLElement;
 
-    readonly coordConstrainedWithinInteractiveBounds: (coord: InputEventCoordinate) => boolean;
+    /**
+     * A boundary constraining the legal coordinates for supported touchstart and mousedown
+     * events.  If not specified, this will be set to `targetRoot`.
+     */
+    readonly inputStartBounds?: RecognitionZoneSource;
+
+    /**
+     * A boundary constraining the maximum range that an ongoing input may travel before it
+     * is forceably canceled.  If not specified, this will be set to `targetRoot`.
+     */
+    readonly maxRoamingBounds?: RecognitionZoneSource;
+
+    /**
+     * A boundary constraining the "safe range" for ongoing touch events.  Events that leave a
+     * safe boundary that did not start outside its respective "padded" bound will be canceled.
+     *
+     * If not specified, this will be based on the active viewport, padded internally by 2px on
+     * all sides.
+     */
+    readonly safeBounds?: RecognitionZoneSource;
+
+    /**
+     * Used to define a "boundary" slightly more constrained than `safeBounds`.  Events that
+     * start within this pixel range from a safe bound will disable that bound for the duration
+     * of its corresponding input sequence.  May be a number or an array of 1, 2, or 4 numbers,
+     * as with CSS styling.
+     *
+     * If not specified, this will default to a padding of 3px inside the standard safeBounds.
+     */
+    readonly safeBoundPadding?: number | number[];
+  }
+
+  export interface FinalizedGestureRecognizerConfiguration extends Nonoptional<GestureRecognizerConfiguration> {
```

Comment:
Is there a real need for a separate interface? It seems like it adds a lot of boilerplate.

Reply:
It's just a little too OO for my comfort -- why not just make `paddedSafeBounds` optional on the input object as you say with corresponding comments? It's all internal anyway so it's not like we are making API contracts.

Reply:
I've made an adjustment in this regard; are the recent changes enough to address this concern?

Reply:
Yes thanks